cfme/infrastructure/provider/__init__.py

```diff
@@ -286,10 +292,74 @@ def hosts(self):
         return result
 
 
+@navigator.register(Provider, 'All')
+class All(CFMENavigateStep):
+    prerequisite = NavigateToAttribute('appliance', 'LoggedIn')
+
+    def step(self):
+        from cfme.web_ui.menu import nav
+        nav._nav_to_fn('Compute', 'Infrastructure', 'Providers')(None)
+
+
+@navigator.register(Provider, 'Add')
+class Add(CFMENavigateStep):
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        cfg_btn('Add a New Infrastructure Provider')
+
+
+@navigator.register(Provider, 'Discover')
+class Discover(CFMENavigateStep):
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        cfg_btn('Discover Infrastructure Providers')
+
+
+@navigator.register(Provider, 'Details')
+class Details(CFMENavigateStep):
+    def prerequisite(self):
+        NavigateToSibling('All')
+        # Make sure no quadicons are selected and default to grid view
+        # Can we assume these two things at this level? (comment to be removed before merge)
+        # If this should be actively set, a utility function will be needed to shorten this
+        tb.select("Grid View")
```

Comment:
default to grid view is definitely required and if these steps go into the previous destination I think we don't need to worry about the selection. As it doesn't remember it from page to page and it will have "just" done a navigation. Though I do see your point that if we DO have an am_i_here, we may need a resetter. Perhaps we can insert a resetter into the flow, only to be run on certain occasions, or always to be run even if am_i_here is true.